packages/core/src/render3/instructions/shared.ts

```diff
@@ -863,28 +863,39 @@ export function createTNode(
 
 
 function generatePropertyAliases(
-    inputAliasMap: {[publicName: string]: string}, directiveDefIdx: number,
-    propStore: PropertyAliases|null): PropertyAliases|null {
+    inputAliasMap: {[publicName: string]: string}, directiveIndex: number,
+    propStore: PropertyAliases|null,
+    aliasMap: {[internalName: string]: string}|null): PropertyAliases|null {
```

Comment:
nit: I think it'd be great to have a more descriptive name here:
```suggestion
    hostDirectiveInputAliases: {[internalName: string]: string}|null): PropertyAliases|null {
```

Reply:
This function would also benefit from having a JSDocs that explains the processing that happens inside.

Reply:
I added a JSDoc and some more explanation in the comments, but I don't agree about the new name since the mapping is used both for inputs and outputs.